doc/cephadm/upgrade.rst

```diff
@@ -188,3 +188,101 @@ you need. For example, the following command upgrades to a development build:
+
+.. prompt:: bash #
+
+  ceph orch upgrade start --image <image-name> --daemon-types mgr,mon --hosts host1,host2
+
+Example, specifying services:
+
+.. prompt:: bash #
+
+  ceph orch upgrade start --image <image-name> --services rgw.example1,rgw.example2
+
+.. note::
+
+   cephadm strictly enforces an order to the upgrade of daemons that is still present
+   in staggered upgrade scenarios. The current upgrade ordering is
+   ``mgr -> mon -> crash -> osd -> mds -> rgw -> rbd-mirror -> cephfs-mirror -> iscsi -> nfs``.
+   If you attempt to specify parameters that would force us to upgrade daemons out of
+   order, the upgrade command will block and inform you of which daemons earlier
+   in the upgrade order will be missed using your parameters.
+
+.. note::
+
+   In order to verify the upgrade order will not be broken, upgrade commands with
+   extra limiting parameters must first validate the options before beginning the
+   upgrade. This may require pulling the image you are trying to upgrade to. Do
+   not be surprised if the upgrade start command takes a while to return when limiting
+   parameters are provided.
+
+.. note::
+
+   In staggered upgrade scenarios (when a limiting parameter is provide) monitoring
+   stack daemons such as prometheus or node-exporter are "upgraded" (if you are
+   not using the default image or if the default does not change between upgrade
+   versions it's actually just a redeploy) after the mgr daemons are finished
+   upgrading. Do not be surprised if you attempt to upgrade mgr daemons and it
+   takes longer than expected as it may just be upgrading monitoring stack daemons
+   as well.
+
+Upgrading to a version that supports staggered upgrade from one that doesn't
+----------------------------------------------------------------------------
+
+While upgrading from a version that already supports staggered upgrades the process
+simply requires providing the necessary arguments. However, if you wish to upgrade
+TO a version that supports staggered upgrade from one that does not, there is a
+workaround that can be used. It requires first manually upgrading the mgr daemons
+and then passing the limiting parameters normally from that point on.
+
+.. warning::
+  Make sure you have multiple running mgr daemons before attempting this procedure.
+
+To start with, determine which mgr is your active one and which are standby. This
+can be done in a variety of ways such as looking at the ``ceph -s`` output. Then,
+manually upgrade each standby mgr daemon with:
+
+.. prompt:: bash #
+
+  ceph orch daemon redeploy mgr.example1.abcdef --image <new-image-name>
+
+.. note::
+
+   If you are on a very early version of cephadm (early Octopus) the ``orch daemon redeploy``
+   command may not have the ``--image`` flag. In that case, you must manually set the
+   mgr container image ``ceph config set mgr container_image <new-image-name>`` and then
+   redeploy the mgr ``ceph orch daemon redeploy mgr.example1.abcdef``
+
```

Comment:
ditto for these and below instead of `mgr` that aren't part of example commands.